rfcs/0000-top-level-assumptions.md

```diff
@@ -0,0 +1,327 @@
+| `setPublicClassFields` | When using computed public class fields, use `[[Set]]` semantics (i.e. use an assignment) instead of `[[Define]]` | `loose` in `proposal-class-properties` | [🔗](https://babeljs.io/repl#?browsers=chrome%2070&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAECC0G8BQ1oDMCWBTEATaAvNAIwDcSAvkA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Cenv&prettier=false&targets=&version=7.10.1&externalPlugins=) |
+| `privateFieldsAsProperties` | Instead of storing private fields and methods using a `WeakMap` or a `WeakSet`, define them as own non-enumerable properties of the class instance | `loose` in `proposal-class-properties` _(not documented)_, in `proposal-private-methods` and `proposal-private-property-in-object` | [🔗](https://babeljs.io/repl#?browsers=chrome%2070&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAECC0G8BQ1oGIBmBLApiAJtALzQCMA3CtFWgLY4AuAFgPb4AUAlIlQJABOjAK78AdtGZYIAOky4ClVAF8kSoA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Cenv&prettier=false&targets=&version=7.10.1&externalPlugins=) | 3 |
+| `superAsFunctionCall` | When this option is enabled, `super(arg1)` will be transpiled to `BaseClass.call(this, arg1)`. This means that it won't work with native classes or with built-ins, but only with compiled classes or ES5 constructors | `loose` in `transform-classes` | |
+| `inheritAsObjectCreate` | Use the `inheritsLoose` helper instead of `inherits` for derived classes | `loose` in `transform-classes` | |
+| `noDocumentAll` | When compiling the `??` and `?.` operators, assume that they are never used with `document.all` and thus `== null` is safe | `loose` in `proposal-optional-chaining` and `proposal-nullish-coalescing-operator` | [🔗](https://babeljs.io/repl#?browsers=chrome%2070&build=&builtIns=false&spec=false&loose=false&code_lz=IYfgdARg3AUMAEIT2kA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Cenv&prettier=false&targets=&version=7.10.1&externalPlugins=) | 4 |
+| `setModuleMeta` | When compiling ESM to CJS, Babel defined a non-enumerable, non-witable, non-configurable `__esModule` property on the `exports` object. When this option is enabled, that property is set using a simple assignment | `loose` in `transform-modules-commonjs` |
+| `copyReexports` | When re-exporting an imported value, assume that it's value doesn't change and set it with a simple assignment | `loose` in `transform-modules-commonjs`, `transform-modules-umd`, `transform-modules-amd` _(not documented)_ | | |
+| `newableArrowFunctions` | Do not check that arrow functions are never instantiated using `new` | `spec` in `transform-arrow-functions` | | 5 |
```

Comment:
In the future we might also want to introduce this for object/class methods, since they also shouldn't be newable according to the spec.

We could either name this `newableFunctions`, or decide to keep them separate (`newableArrowFunctions` and `newableMethods`).